service/dap/server.go

```diff
@@ -1511,10 +1559,105 @@ func (s *Server) onReverseContinueRequest(request *dap.ReverseContinueRequest) {
 	s.sendNotYetImplementedErrorResponse(request.Request)
 }
 
-// onSetVariableRequest sends a not-yet-implemented error response.
-// Capability 'supportsSetVariable' is not set 'initialize' response.
-func (s *Server) onSetVariableRequest(request *dap.SetVariableRequest) { // TODO V0
-	s.sendNotYetImplementedErrorResponse(request.Request)
+func variableByName(v *proc.Variable, name string) (*proc.Variable, bool) {
+	if v == nil {
+		return nil, false
+	}
+	if v.Name == name {
+		return v, true
+	}
+
+	for _, child := range v.Children {
+		if child.Name == name {
+			return &child, true
+		}
+		// No need to look into the child's children - DAP clients are expected
+		// to provide the variable reference number, so the name is the name of
+		// the variable's direct child.
+	}
+	return nil, false
+}
+
+// nameToEvaluateName finds the variable of the specified name
+// from the given variable and its children, and returns its
+// evaluateName.
+func (s *Server) nameToEvaluateName(v *fullyQualifiedVariable, name string) (string, error) {
+	if v.Name == name && !v.isScope { // will this ever happen?
```

Comment:
I think per dap spec, you are not supposed to consider the variable (aka "container") itself, just its children. "Name - The name of the variable in the container". I think this is because the leaf variables won't have a reference, so the only way to identify them would be to express them as variables in a parent container with some reference.

Reply:
That's what I am thinking too. Do you want me to remove this?

Reply:
removed.